fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    const getUserById = async (): Promise<void> => {
+      try {
+        const result = await fetch(
+          `https://jsonplaceholder.typicode.com/users/${userId}`
+        );
+        if (!result.ok)
+          throw new Error('could not fetch userData from clicked user');
+        const formattedResult = (await result.json()) as Promise<User>;
```

Comment:
The `awaited` result is just `User`. `result.json()` is a Promise, but once it's awaited the promise gets removed.